spec/factories/events.rb

```diff
@@ -20,7 +20,56 @@
 
 FactoryBot.define do
   factory :event do
-    name { %w[pull_request review review_comment].sample }
-    data { Faker::Json.shallow_json(width: 3) }
+    transient do
+      action {}
+      pull_request_payload {}
+    end
+
+    data { build :check_run_payload }
+
+    name { 'check_run' }
+
+    association :project, factory: :project
+
+    after(:build) do |event, factory|
+      event.data['action'] = factory.action if factory.action
+      event.data['pull_request'] = factory.pull_request_payload if factory.pull_request_payload
+    end
+
+    factory :event_pull_request do
+      transient do
+        created_at { Faker::Date.between(from: 1.month.ago, to: Date.yesterday) }
+        updated_at { Faker::Date.between(from: created_at, to: Date.yesterday) }
```

Comment:
`to: Time.current` maybe? to prevent the case `created_at == updated_at == Date.yesterday` ??